### PR TITLE
Build grpc on Windows using OpenSSL and zlib from conan.io

### PR DESCRIPTION
**Pull request by KindDragon**
What do you think about this changes?

### CODE REVIEW

File: vsprojects/README.md [hunk 1]
```diff
@@ -90,17 +90,12 @@ $ cmake -G "Visual Studio 12 2013"
...
-#Building using CMake (with BoringSSL)
-1. Install [Active State Perl](http://www.activestate.com/activeperl/) (`choco install activeperl`)
-2. Install [Ninja](https://ninja-build.org/) (`choco install ninja`)
-2. Install [Go](https://golang.org/dl/) (`choco install golang`)
-3. Install [yasm](http://yasm.tortall.net/) and add it to `PATH` (`choco install yasm`)
-4. Update boringssl sumbodule to `master`
-5. Run this commads in grpc directory:
+#Building using CMake
+1. Install [Conan.io](https://www.conan.io/downloads)
+2. Run this commads in grpc directory:
```
Comment (nicolasnoble) [inline vsprojects/README.md]: There's a 'n' missing in 'commands'.

Comment (nicolasnoble) [inline vsprojects/README.md]: Actually, that's not your fault, I realize :-)

**Comment by nicolasnoble** [comment]
So, if conan isn't integrated into cmake, and we're just going to run commands from cmake to run conan, couldn't we already do that with nuget instead ?

Not sure if using nuget packages would be easier or worse than conan however.

**Comment by KindDragon** [comment]
> So, if conan isn't integrated into cmake, and we're just going to run commands from cmake to run conan, couldn't we already do that with nuget instead ?

I tried nuget first, but it's really hard to add nuget dependepcy to cmake. CMake cannot add .target file to vsiual studo project. With conan I can just do [`find_package`](http://conanio.readthedocs.io/en/latest/integrations/cmake.html):
```
elseif("${gRPC_SSL_PROVIDER}" STREQUAL "conan")
  find_package(OpenSSL REQUIRED)
  set(_gRPC_SSL_LIBRARIES "${CONAN_LIBS_OPENSSL}")
endif()
```

They just often call `conan install` not from cmake, but conan has good integration with cmake.

**Comment by nicolasnoble** [comment]
Ah, I didn't spot that in the change. Alright then, thanks :-)

File: INSTALL.md [hunk 2]
```diff
@@ -61,6 +61,17 @@ gRPC C Core library.
 There are several ways to build under Windows, of varying complexity depending
 on experience with the tools involved.
 
+###CMake and Conan
```
Comment (KindDragon) [inline INSTALL.md]: @nicolasnoble Can you try this instruction? Will it work for you?

**Comment by KindDragon** [comment]
PR is ready to merge. I do not know if I changed template correctly

**Comment by KindDragon** [comment]
Jenkins pls stop :-)

**Comment by jtattermusch** [comment]
I am skeptical. Changes to the way we build things have the potential to break lots of stuff, need thorough testing and potentially involve changes to our testing infrastructure (which is costly as it needs to be done manually). So I'd expect that the pull request description would contain summary of what you are trying to achieve and especially *why* it is going to be useful (and also what are the drawbacks - there are always some). Without a good description of the PR I have very low confidence in the changes you are suggesting.

**Comment by KindDragon** [comment]
I just tried to simplify build for Windows developers (which is not against installing conan.io).
If you not ok with that changes just close PR

**Comment by jtattermusch** [comment]
@KindDragon  I didn't mean that this PR should be closed, I just wanted to point out what are some important things that this PR is missing. I am also not saying this does not simplify the build for Windows developers (simplification would be great), but I don't quite understand how exactly is the build experience simplified what are the tradeoffs. In any case, thanks for the PR.

**Comment by oleiade** [comment]
For the record, I found this PR very useful, and helped me building grpc on Windows. Which as of today's `master` branch does not work out of the box.

**Pull request closed without merging**